### ISSUE TITLE
MCP23017 wrong "startup logic"

### ISSUE THREAD

**Issue by bilbolodz** [domoticz/domoticz#2616]
I'm trying to use "built in" MCP23017 module.
All is working fine when device is declared with option "Inverted GPIO" turning on/off device in Domoticz changes state on MCP23017 chip. Also after Domoticz startup correct state of Domoticz switches is restored to hardware.
But when "Inverted GPIO" option is turned off it's getting strange: switching on/off of the switch in correctly "reproduced to hardware" BUT after startup domoticz restores hardware to state "as option Inverted GPIO was on".

To summarize:

Inverted GPIO ON:
Switch ON -> HW: 0; Swich OFF -> HW: 1;Switch ON -> HW:0; Turn off RPI;After Domoticz startup HW become to 0 because last swicth state was ON - FINE

Inverted GPIO OFF:
Switch ON -> HW: 1; Swich OFF -> HW: 0;Switch ON -> HW:1; Turn off RPI;After Domoticz startup HW become to 0 BUT SHOULD be 1 because last swicth state was ON - ERROR!!!
 
Tested with stable version and latest beta.

**Comment by rwaaren** [comment]
Issue closed because of no activity for a very long time. If you believe this is incorrect please update the issue with the reason why and with the required action to get this closed as no one seems to be able to help.
